mailpoet/tests/DataGenerator/Generators/WooCommercePastRevenues.php

```diff
@@ -268,59 +285,66 @@ private function getBatchLog($dataType, $generatedCount) {
-    ORM::rawExecute("ALTER TABLE `" . SubscriberSegment::$_table . "` ENABLE KEYS");
-    ORM::rawExecute("ALTER TABLE `" . $this->entityManager->getClassMetadata(NewsletterLinkEntity::class)->getTableName() . "` ENABLE KEYS");
-    ORM::rawExecute("ALTER TABLE `" . ScheduledTask::$_table . "` ENABLE KEYS");
-    ORM::rawExecute("ALTER TABLE `" . ScheduledTaskSubscriber::$_table . "` ENABLE KEYS");
-    ORM::rawExecute("ALTER TABLE `" . SendingQueue::$_table . "` ENABLE KEYS");
-    ORM::rawExecute("ALTER TABLE `" . $this->entityManager->getClassMetadata(StatisticsOpenEntity::class)->getTableName() . "` ENABLE KEYS");
-    ORM::rawExecute("ALTER TABLE `" . $this->entityManager->getClassMetadata(StatisticsClickEntity::class)->getTableName() . "` ENABLE KEYS");
-    ORM::rawExecute("SET UNIQUE_CHECKS = 1;");
+    $connection->executeStatement("ALTER TABLE `{$prefix}posts` ENABLE KEYS");
+    $connection->executeStatement("ALTER TABLE `{$prefix}postmeta` DISABLE KEYS");
+    $connection->executeStatement("ALTER TABLE `" . $this->entityManager->getClassMetadata(SubscriberEntity::class)->getTableName() . "` ENABLE KEYS");
+    $connection->executeStatement("ALTER TABLE `" . $this->entityManager->getClassMetadata(SubscriberSegmentEntity::class)->getTableName() . "` ENABLE KEYS");
+    $connection->executeStatement("ALTER TABLE `" . $this->entityManager->getClassMetadata(NewsletterLinkEntity::class)->getTableName() . "` ENABLE KEYS");
+    $connection->executeStatement("ALTER TABLE `" . $this->entityManager->getClassMetadata(ScheduledTaskEntity::class)->getTableName() . "` ENABLE KEYS");
+    $connection->executeStatement("ALTER TABLE `" . $this->entityManager->getClassMetadata(ScheduledTaskSubscriberEntity::class)->getTableName() . "` ENABLE KEYS");
+    $connection->executeStatement("ALTER TABLE `" . $this->entityManager->getClassMetadata(SendingQueueEntity::class)->getTableName() . "` ENABLE KEYS");
+    $connection->executeStatement("ALTER TABLE `" . $this->entityManager->getClassMetadata(StatisticsOpenEntity::class)->getTableName() . "` ENABLE KEYS");
+    $connection->executeStatement("ALTER TABLE `" . $this->entityManager->getClassMetadata(StatisticsClickEntity::class)->getTableName() . "` ENABLE KEYS");
+    $connection->executeStatement("SET UNIQUE_CHECKS = 1;");
   }
 
-  private function createSubscriber($email, $lastName, $createdAtDate, \MailPoet\Models\Segment $segment, $status = Subscriber::STATUS_SUBSCRIBED) {
-    $subscriber = Subscriber::createOrUpdate([
-      'email' => $email,
-      'status' => $status,
-      'last_name' => $lastName,
-      'created_at' => $createdAtDate,
-    ]);
-    $subscriber->save();
-    $segment->addSubscriber($subscriber->id);
+  private function createSubscriber(string $email, string $lastName, string $createdAtDate, SegmentEntity $segment, $status = SubscriberEntity::STATUS_SUBSCRIBED): SubscriberEntity {
+    $subscriber = new SubscriberEntity();
```

Comment:
`MailPoet\Test\DataFactories\Subscriber` can be used to make the creation easier

Reply:
I don't see any difference in this particular case in terms of making the creation easier, but at the same time, I'm not against it. I implemented this change in a new commit.